ui/raidboss/data/06-ew/dungeon/the_tower_of_zot.ts

```diff
@@ -34,56 +37,93 @@ const triggerSet: TriggerSet<Data> = {
       // 631B = Transmute Blizzard III
       // 631C = Transmute Thunder III
       // 631D = Transmute Bio III
-      netRegex: NetRegexes.startsUsing({ id: ['629A', '631[BCD]'], source: 'Minduruva', capture: false }),
-      netRegexDe: NetRegexes.startsUsing({ id: ['629A', '631[BCD]'], source: 'Rug', capture: false }),
-      netRegexFr: NetRegexes.startsUsing({ id: ['629A', '631[BCD]'], source: 'Anabella', capture: false }),
-      netRegexJa: NetRegexes.startsUsing({ id: ['629A', '631[BCD]'], source: 'ラグ', capture: false }),
-      // FIXME: if this is `run` then data.orbCount has an off-by-one (one less) count in the emulator.
-      // `run` must happen synchronously before other triggers if the trigger is not asynchronous.
-      // It's possible this is a general raidboss bug as well, but it is untested.
-      preRun: (data) => data.orbCount++,
+      netRegex: NetRegexes.startsUsing({ id: ['629A', '631[BCD]'], source: 'Minduruva' }),
+      netRegexDe: NetRegexes.startsUsing({ id: ['629A', '631[BCD]'], source: 'Rug' }),
+      netRegexFr: NetRegexes.startsUsing({ id: ['629A', '631[BCD]'], source: 'Anabella' }),
+      netRegexJa: NetRegexes.startsUsing({ id: ['629A', '631[BCD]'], source: 'ラグ' }),
+      run: (data, matches) => {
+        data.orbCount++;
+
+        // We only expect one of these at once
+        if (matches.id === '629A')
+          data.orbs.set('Fire', data.orbCount);
+        else if (matches.id === '631D')
+          data.orbs.set('Bio', data.orbCount);
+      },
     },
     {
-      id: 'Zot Minduruva Transmute Fire III',
+      id: 'Zot Minduruva Manusya III',
       type: 'StartsUsing',
-      netRegex: NetRegexes.startsUsing({ id: '629A', source: 'Minduruva', capture: false }),
-      netRegexDe: NetRegexes.startsUsing({ id: '629A', source: 'Rug', capture: false }),
-      netRegexFr: NetRegexes.startsUsing({ id: '629A', source: 'Anabella', capture: false }),
-      netRegexJa: NetRegexes.startsUsing({ id: '629A', source: 'ラグ', capture: false }),
-      durationSeconds: 13,
-      // These are info so that any Under/Behind from Fire III / Bio III above take precedence.
-      // But, sometimes the run from Bio III -> Transmute Fire III is tight so warn ahead of
-      // time which orb the player needs to run to.
-      infoText: (data, _matches, output) => output.text!({ num: data.orbCount }),
+      // 6291 = Manusya Fire III
+      // 6292 = Manusya Blizzard III
+      // 6293 = Manusya Thunder III
+      // 6294 = Manusya Bio III
+      netRegex: NetRegexes.startsUsing({ id: ['629[1-4]'], source: 'Minduruva' }),
+      netRegexDe: NetRegexes.startsUsing({ id: ['629[1-4]'], source: 'Rug' }),
+      netRegexFr: NetRegexes.startsUsing({ id: ['629[1-4]'], source: 'Anabella' }),
+      netRegexJa: NetRegexes.startsUsing({ id: ['629[1-4]'], source: 'ラグ' }),
+      durationSeconds: (data) => {
+        if (data.orbs.size > 0)
+          return 7.40;
+
+        return 3.70;
```

Comment:
These are highly specific numbers!

Reply:
For what it's worth, duration is always 0.3 shorter than the actual cast time.  I think for things that that need to be resolved immediately (e.g. get behind right now), I'd just use the normal duration (e.g. theoretically just `return;` will use the default?) rather than specifying something different here.

For the longer one, maybe check times from a log between when this would trigger and when the mechanic would go off, and do something around that length?

Reply:
I checked my log files I have, and it looks like the 1st orb resolves 8 seconds after the Manusya cast, while the 2nd orb resolves after 12 seconds. I just make this use the default duration for no mechanic overlap and 12 seconds if there is an overlap. We could also instead pull out which orb is overlapping and then make the duration 8 or 12 based on that number, but its a bit wonky to do that.